meshio/gmsh/main.py

```diff
@@ -7,7 +7,7 @@
 
 # Some mesh files out there have the version specified as version "2" when it really is
 # "2.2". Same with "4" vs "4.1".
-_readers = {"2": _gmsh22, "2.2": _gmsh22, "4.0": _gmsh40, "4": _gmsh41, "4.1": _gmsh41}
+_readers = {"2": _gmsh22, "2.2": _gmsh22, "4.0": _gmsh40, "4": _gmsh40, "4.1": _gmsh41}
```

Comment:
Unfortunately, gmsh didn't change between 4.0 and 4.1 in that "4" can mean both. So, this change needs to be reverted.

Reply:
In the source code of gmsh, the mesh version is read as a double and the reader is filled with comparisons like `if(version >= 4.1)` so  '4' should imply gmsh40 format. In addition, if you write a mesh with a recent gmsh (I'm using 4.5), the `MeshFormat` block will always contain '4.1'. Only if you specify `gmsh -format msh40` on the command line, will `MeshFormat` contain '4', so that seems to be quite consistent? I included this change because I wanted to test this PR on a mesh file generated with `gmsh -format msh40`. meshio failed to read this initially, and only by switching to the gmsh40 did it work. The error was:
```
  File "meshio/meshio/gmsh/_gmsh41.py", line 115, in _read_entities
    (num_physicals,) = fromfile(f, c_size_t, 1)
ValueError: not enough values to unpack (expected 1, got 0)
```

Reply:
>  so '4' should imply gmsh40 format.

See [here](https://github.com/nschloe/meshio/blob/master/meshio/gmsh/main.py#L9). We've have problems with "4" meaning "4.0" in the past. I remember we've changed that before, and it's always going back and forth, which is why I'm reluctant to change here. Honestly if gmsh writes "4" and mean "4.0", than that's a bug in gmsh.

Since this creates some confusion, I'm inclined to disallow format "4" altogether and always error out.

>  gmsh -format msh40

Wait, _this_ leads to the 4.1 reader? That looks like a bug.